cpp/src/io/parquet/reader_impl.cpp

```diff
@@ -210,14 +211,27 @@ void reader::impl::decode_page_data(read_mode mode, size_t skip_rows, size_t num
     }
   }
 
+  // Create an empty device vector to store the initial str offset for large string columns from for
+  // string decoders.
+  auto initial_str_offsets = rmm::device_uvector<size_t>{0, _stream, _mr};
+
   pass.chunks.host_to_device_async(_stream);
   chunk_nested_valids.host_to_device_async(_stream);
   chunk_nested_data.host_to_device_async(_stream);
-  if (has_strings) { chunk_nested_str_data.host_to_device_async(_stream); }
+  if (has_strings) {
+    // Initialize the initial string offsets vector.
+    initial_str_offsets = cudf::detail::make_device_uvector_async(
+      std::vector<size_t>(_input_columns.size(), std::numeric_limits<size_t>::max()), _stream, _mr);
+    chunk_nested_str_data.host_to_device_async(_stream);
+  }
 
   // create this before we fork streams
   kernel_error error_code(_stream);
 
+  // create a device span of initial string offsets vector
+  auto initial_str_offsets_span =
+    cudf::device_span<size_t>{initial_str_offsets.begin(), initial_str_offsets.size()};
```

Comment:
Can we avoid declaring a span here and pass `initial_str_offsets`? device_span can be implicitly constructed from a device_uvector